### PR TITLE
Add support for DXArchive8.

### PR DESCRIPTION
**Pull request by SlawekNowy**
Not finished here beacuse cannot proceed without password. 
DxLib is open source and so are DxaDecode and DxaEncode.

### CODE REVIEW

File: ArcFormats/DxLib/ArcDX.cs [hunk 1]
```diff
@@ -313,12 +313,15 @@ DxHeader ReadArcHeaderV6 (ArcView file, int version, byte[] key)
 
         internal static void Decrypt (byte[] data, int index, int count, long offset, byte[] key)
         {
-            int key_pos = (int)(offset % key.Length);
-            for (int i = 0; i < count; ++i)
+            if (key.Length !=0)
```
Comment (crskycode) [inline ArcFormats/DxLib/ArcDX.cs:316]: Just return
```csharp
if (key.Length == 0)
    return;
```

File: ArcFormats/Properties/Settings.Designer.cs
Comment (crskycode) [inline ArcFormats/Properties/Settings.Designer.cs:1]: These changes of comment do not need to be commit.

File: GameRes/ArcView.cs [hunk 2]
```diff
@@ -33,7 +33,8 @@ namespace GameRes
 {
     public static class Encodings
     {
-        public static readonly Encoding cp932 = Encoding.GetEncoding (932);
+        public static readonly Encoding cp932 = Encoding.GetEncoding(932);
+        public static readonly Encoding ASCII = Encoding.ASCII;
```
Comment (crskycode) [inline GameRes/ArcView.cs:37]: unnecessary

**Review by crskycode — COMMENTED** [review]
There are some minor issues, please do not commit unnecessary changes and keep the code style consistent.

**Comment by SlawekNowy** [comment]
Turns out I need to fix BitStreams implementations to support arbitrary number of bits.
Huffman compression needs up to 64 bits here. MSB supports 24 bits, LSB 32 bits.

**Comment by SlawekNowy** [comment]
Ok. Implementation is done. Do I need to update the schema with passwords for DXA8?

**Comment by SlawekNowy** [comment]
Actually hold on. I have data corruption on non-ASCII filepaths.

**Comment by crskycode** [comment]
Thanks for your contribution! As I've been busy lately, it may be a few days before I have time to review the code.

**Review by crskycode — COMMENTED** [review]
Reviewed

**Pull request merged**